kubernetes_asyncio/config/kube_config.py

```diff
@@ -313,11 +313,33 @@ async def load_from_exec_plugin(self):
         try:
             if hasattr(self, 'exec_plugin_expiry') and not _is_expired(self.exec_plugin_expiry):
                 return True
+            base_path = self._get_base_path(self._cluster.path)
             status = await ExecProvider(self._user['exec']).run()
-            if 'token' not in status:
-                logging.error('exec: missing token field in plugin output')
+            if 'token' in status:
+                self.token = "Bearer %s" % status['token']
+            elif 'clientCertificateData' in status:
+                # https://kubernetes.io/docs/reference/access-authn-authz/authentication/#input-and-output-formats
+                # Plugin has provided certificates instead of a token.
+                if 'clientKeyData' not in status:
+                    logging.error('exec: missing clientKeyData field in '
+                                  'plugin output')
+                    return None
+                self.cert_file = FileOrData(
+                    status, None,
+                    data_key_name='clientCertificateData',
+                    file_base_path=base_path,
+                    base64_file_content=False,
+                    temp_file_path=self._temp_file_path).as_file()
+                self.key_file = FileOrData(
+                    status, None,
+                    data_key_name='clientKeyData',
+                    file_base_path=base_path,
+                    base64_file_content=False,
+                    temp_file_path=self._temp_file_path).as_file()
+            else:
+                logging.error('exec: missing token or clientCertificateData '
+                              'field in plugin output')
                 return None
-            self.token = "Bearer %s" % status['token']
             if 'expirationTimestamp' in status:
```

Comment:
expirationTimestamp is related to the token authentication, could you move it there?

Reply:
@tomplus I moved `expirationTimestamp` next to the the token authentication block.

FYI, I initially kept it where separately to make the code [similar to the sync-client](https://github.com/kubernetes-client/python/blob/76ec6f5ce5572f66d82717e296480dc42584f0b7/kubernetes/base/config/kube_config.py#L516-L517) (I think it would make it simpler to sync changes between the 2 implementations)

Reply:
Right, I should have checked the sync client :+1: I think we can leave it as is.